### PR TITLE
generate function for asking id

### PR DESCRIPTION
**Pull request by icerove**
generate asking question askForId to ask user whether okay to expose their account id when typing account id with command. This is for mixpanel project of identity management 
change config 'default' with 'testnet'

### CODE REVIEW

File: utils/eventtracking.js [hunk 1]
```diff
@@ -109,6 +109,53 @@ const getEventTrackingConsent = async () => {
...
+    const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+    });
+    try {
+        for (let attempts = 0; attempts < 10; attempts++) {
+            const answer = await new Promise((resolve) => {
+                rl.question(
+                    chalk`We would like to help with your development journey with NEAR.` +
+                        chalk` We will ask you to expose your account Id while using command. ` +
```
Comment (mikedotexe) [inline utils/eventtracking.js:122]: Do you think we could replace "expose" with "share" here and the next line?

Comment (icerove) [reply utils/eventtracking.js:122]: yes, definitely, good idea, thanks

**Comment by mikedotexe** [comment]
It keeps asking me the same question after I say "no, thank you" by typing **n**
I was using this command:
`./bin/near state mike.testnet`

Screen Shot 2020-10-27 at 2 37 39 PM

**Comment by icerove** [comment]
I will fix this, thanks

**Comment by mikedotexe** [comment]
The tests seem to be failing as it's waiting for a response on this new prompt:

> ++../bin/near state testaccount1603834350.test.near -v
> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
> Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
> The build has been terminated

**Comment by icerove** [comment]
I see, I will fix the test, too. Thanks

**Comment by mikedotexe** [comment]
Sorry, I hate to keep adding comments, but I'm trying all the different ways.
I think if the user says, "no thank you, please don't track me" we should not ask them again if they want to share their account ID.
For instance, if I do `near login` and it prompts me, and I say, "no thank you" and then do `near state` it will ask me if I want to share my account ID.
I think if the answer is no to the first question, the second question should not be asked, but that is not how it behaves now.

**Comment by mikedotexe** [comment]
Unfortunately I cannot see Mixpanel because of a bug on their end, but I've filed a ticket. I think as long as the most recent comment is addressed and it's showing the person's accountId in Mixpanel, good to go!

**Comment by icerove** [comment]
Thank you so much for testing and reviewing for me.
Actually, I consider the no situation here, and when the setting is like this: 
image

File: utils/eventtracking.js [hunk 2]
```diff
@@ -109,6 +124,57 @@ const getEventTrackingConsent = async () => {
...
+        }
+        return false; // If they can't figure it out in this many attempts, just opt out
+    } finally {
+        rl.close();
+    }
+};
+
+const askForId = async (options) => {
+    const shellSettings = settings.getShellSettings();
+    if(shouldTrackID(shellSettings)){
```
Comment (icerove) [inline utils/eventtracking.js:163]: this is when the answer is 'Y' or 'y'

File: utils/eventtracking.js [hunk 3]
```diff
@@ -109,6 +124,57 @@ const getEventTrackingConsent = async () => {
...
+        const id = isGitPod() ? getGitPodUserHash() : shellSettings[TRACKING_SESSION_ID_KEY];
+        await Promise.all([
+            mixpanel.alias(options.accountId, id),
+            mixpanel.people.set_once({account_id: options.accountId})
+        ]);
+    }else if(shouldNOTTrackID(shellSettings)){
```
Comment (icerove) [inline utils/eventtracking.js:169]: this is when answer is 'N' or 'n'

File: utils/eventtracking.js [hunk 4]
```diff
@@ -109,6 +124,57 @@ const getEventTrackingConsent = async () => {
...
+        return;
+    }
+    else{
```
Comment (icerove) [inline utils/eventtracking.js:172]: This is first time to ask for that

**Comment by icerove** [comment]
Would you mind check where I point out again?

**Review by mikedotexe — APPROVED** [review]
awesome!

**Pull request merged**